src/code.cloudfoundry.org/silk/controller/leaser/cidrpool_test.go

```diff
@@ -9,110 +9,202 @@ import (
+					numCheckedBlockPools++
 					_, blockNetwork, _ := net.ParseCIDR(blockDividedCIDR)
-					Expect(overlayNetwork.Contains(blockNetwork.IP)).Should(BeTrue())
+					Expect(oneNetworkContains(networks, blockNetwork.IP)).Should(BeTrue())
 				}
+
+				Expect(numCheckedBlockPools).To(Equal(expectedNumBlockPools))
 			},
-			Entry("when ip is in the start of the cidr range", "10.240.0.0/12", 24),
-			Entry("when ip is in the middle of the cidr range", "10.255.0.0/12", 24),
-			Entry("when ip is in the end of the cidr range", "10.255.255.255/12", 24),
+
+			// /12 = 1048576 ; /24 = 256 ; 1048576 / 256 = 4096 ; 4096 - 1 (first block is excluded) = 4095
+			Entry("when ip is in the start of the cidr range", []string{"10.240.0.0/12"}, 24, 4095),
+			Entry("when ip is in the middle of the cidr range", []string{"10.255.0.0/12"}, 24, 4095),
+			Entry("when ip is in the end of the cidr range", []string{"10.255.255.255/12"}, 24, 4095),
+
+			// /16 x2 = 65536 x2 = 131072 ; /24 = 256 ; 131072 / 256 = 512
+			// 512 - 2 (first block of each network is excluded) = 510
+			Entry("when there is an array of networks", []string{"10.255.0.0/16", "10.250.0.0/16"}, 24, 510),
 		)
 	})
 
 	Describe("SingleIPPoolSize", func() {
 		DescribeTable("returns the number of subnets that can be allocated",
-			func(subnetRange string, subnetMask, expectedSize int) {
+			func(subnetRange []string, subnetMask, expectedSize int) {
 				cidrPool := leaser.NewCIDRPool(subnetRange, subnetMask)
 				Expect(cidrPool.SingleIPPoolSize()).To(Equal(expectedSize))
 			},
-			Entry("when the range is /16 and mask is /25", "10.255.0.0/16", 25, 127),
-			Entry("when the range is /16 and mask is /26", "10.255.0.0/16", 26, 63),
-			Entry("when the range is /16 and mask is /27", "10.255.0.0/16", 27, 31),
+			// /25 = 128, -1 for the first IP which is never allocated.
+			Entry("when the range is /16 and mask is /25", []string{"10.255.0.0/16"}, 25, 127),
+
+			// /26 = 64, -1 for the first IP which is never allocated.
+			Entry("when the range is /16 and mask is /26", []string{"10.255.0.0/16"}, 26, 63),
+
+			// /27 = 32, -1 for the first IP which is never allocated.
+			Entry("when the range is /16 and mask is /27", []string{"10.255.0.0/16"}, 27, 31),
+
+			// /27 = 32, because we get one single IP subnet block from the first overlay network.
+			// -1 for the first IP of the single IP subnet block which is never allocated.
+			Entry("when there are multiple ranges and mask is /27", []string{"10.255.0.0/24", "10.250.0.0/16"}, 27, 31),
 		)
 
-		It("produces valid subnet starting with the first IP of the cidr", func() {
-			overlayCIDR := "10.255.0.0/12"
+		It("produces valid subnet starting with the first IP of the first provided cidr", func() {
+			overlayCIDR := []string{"10.255.0.0/12", "10.20.0.0/16"}
 			subnetMask := 24
-			firstSubnet := "10.240.0.0/24"
+			firstSubnetOfFirstNetwork := "10.240.0.0/24"
+			firstSubnetOfSecondNetwork := "10.20.0.0/24"
```

Comment:
Can you add a comment here about how 10.255.0.0/12 actually starts at 10.240.0.0, since most of the other examples we use are /16, and i had a double take on how this worked until i read closely?